packages/p2p/src/stores/sendbird-store.ts

```diff
@@ -364,100 +395,90 @@ export default class SendbirdStore extends BaseStore {
         };
     }
 
-    setIsUploadComplete(is_upload_complete) {
+    setIsUploadComplete(is_upload_complete: boolean) {
         this.is_upload_complete = is_upload_complete;
     }
 
-    setFileUploadProperties(file_upload_properties) {
+    setFileUploadProperties(file_upload_properties: FileMessage | null) {
         this.file_upload_properties = file_upload_properties;
     }
 
-    progressHandler(progress) {
-        const { order_id } = this.root_store.order_store;
-        this.setIsUploadComplete(progress.loaded === progress.total);
-        if (progress.loaded === progress.total) {
-            setTimeout(() => {
-                if (!this.should_show_chat_on_orders && !order_id) {
-                    this.terminateChatWsConnection();
-                }
-            }, 1000);
-        }
-    }
-
-    sendFile(file) {
+    sendFile(file: File) {
         if (!file) return;
 
-        const params = new this.sendbird_api.FileMessageParams();
-
-        params.file = file;
-        params.fileName = file.name;
-        params.fileSize = file.size;
-        params.mimeType = file.type;
-
-        this.file_upload_properties = this.active_chat_channel.sendFileMessage(
-            params,
-            this.progressHandler,
-            (channel_message, error) => {
-                if (error) {
-                    // eslint-disable-next-line no-console
-                    console.warn(error);
-                } else if (channel_message.channelUrl === this.chat_channel_url) {
+        this.active_chat_channel
+            ?.sendFileMessage({
+                file,
+                fileName: file.name,
+                fileSize: file.size,
+                mimeType: file.type,
+            })
+            .onPending(() => {
+                this.setIsUploadComplete(false);
+            })
+            .onSucceeded(channel_message => {
+                if (channel_message.channelUrl === this.chat_channel_url && channel_message.isFileMessage()) {
                     this.addChannelMessage(convertFromChannelMessage(channel_message));
+                    this.setFileUploadProperties(channel_message);
                 }
-            }
-        );
+                this.setIsUploadComplete(true);
+            })
+            .onFailed(() => {
+                this.setIsUploadComplete(false);
+            });
     }
 
-    sendMessage(message) {
-        const modified_message = message.replace(/^[\r\n]+|[\r\n]+$/g, '');
```

Comment:
Not quite sure why the original author didn't use `message.trim()` here, because it seems like the `regex` just matches and replace `new line` or `return carriage characters`. Will add it to QA checklist to verify this, because the `regex` causes a `security hotspot`, so I decided to replace it with `.trim()`.